spack patches for gcc installation (version 4.9.x)

I was trying to install `gcc` and see following : 

```bash
○ → spack purge -a
○ → spack install -v gcc@4.9.4 %gcc@4.4.7        #same for 4.9.3
....
==> Trying to fetch from http://ftp.gnu.org/gnu/gcc/gcc-4.9.4/gcc-4.9.4.tar.bz2
######################################################################## 100.0%
==> Staging archive: /some-gpfs-path/spack/var/spack/stage/gcc-4.9.4-ybvygharx7lyngmbsgb224cqm65jinib/gcc-4.9.4.tar.bz2
==> Created stage in /some-gpfs-path/spack/var/spack/stage/gcc-4.9.4-ybvygharx7lyngmbsgb224cqm65jinib
Reversed (or previously applied) patch detected!  Assume -R? [n]
```

I haven't looked at which `patches` but wondering if those are backported or something else.

And for `5.2.0`

```bash

$ spack install -v gcc@5.2.0 %gcc@4.4.7

libtool: compile:  /tmp/kumbhar/spack-stage/spack-stage-MgPXZ9/spack-build/./gcc/xgcc -B/tmp/kumbhar/spack-stage/spack-stage-MgPXZ9/spack-build/./gcc/ -B/gpfs/bbp.cscs.ch/home/kumbhar/workarena/systems/lugviz/softwares/sources/spack/opt/spack/linux-redhatenterpriseserver6-x86_64/gcc-4.4.7/gcc-5.2.0-3lxzv5lb6wttupnukbfrztgipznktc3a/x86_64-unknown-linux
....
/gpfs/bbp.cscs.ch/home/kumbhar/workarena/systems/lugviz/softwares/sources/spack/var/spack/stage/gcc-5.2.0-3lxzv5lb6wttupnukbfrztgipznktc3a/gcc-5.2.0/libgfortran/generated/cshift1_4.c  -fPIC -DPIC -o .libs/cshift1_4.o
./libtool: line 1158: /tmp/kumbhar/spack-stage/spack-stage-MgPXZ9/spack-build/./gcc/xgcc: No such file or directory
make[3]: *** [cshift1_4.lo] Error 1
make[3]: *** Waiting for unfinished jobs....
```
I haven't been building `gcc` myself for long time....

--- a/var/spack/repos/builtin/packages/gcc/package.py
+++ b/var/spack/repos/builtin/packages/gcc/package.py
@@ -58,7 +58,7 @@ class Gcc(Package):
         provides('golang', when='@4.7.1:')
 
     patch('piclibs.patch', when='+piclibs')
-    patch('gcc-backport.patch', when='@4.7:5.3')
+    patch('gcc-backport.patch', when='@4.7:4.9.2,5:5.3')
 
     def install(self, spec, prefix):
         # libjava/configure needs a minor fix to install into spack paths.

